### DISCUSSION TITLE
A global message or alert in orchard core

### DISCUSSION THREAD

**Discussion by TaylorEdm** [OrchardCMS/OrchardCore#8366] [Q&A] [answered]
Currently, I am attempting to handle a global alert or message section for a custom orchard core theme. I am struggling with the concepts with Orchard Core so I would like someone to tell me if I need to write a shape, widget, zone, or what?

Requirements: 
1. Located in a custom theme layout (display on all public pages)
2. Be editable inside of the admin section like any other content

Reference: https://github.com/OrchardCMS/OrchardCore/discussions/7334

**Comment by hishamco** [comment]
What's the wrong of using the current notification APIs?

**Reply by TaylorEdm** [reply]
Haven't heard about them. Do you have a link to it?

**Reply by TaylorEdm** [reply]
So I looked into the notification workflow and I don't think that is what I need. This would be more of a global setting where everyone regardless of if they are logged in or not would get a notification.

**Reply by hishamco** [reply]
https://github.com/OrchardCMS/OrchardCore/tree/dev/src/OrchardCore/OrchardCore.DisplayManagement/Notify

There's a messages zone too that shows all the notifications from the system

**Comment by deanmarcussen** [comment, marked as answer]
The concept of the Notifer will not work here as a notification message is direct at a single user via the cookie.

I would use a layer widget for this. Found in the Design menu / Widgets

**Reply by TaylorEdm** [reply]
I can send you 50 bucks if you can hop on a meeting and show me how this is done with it being localized.

**Reply by hishamco** [reply]
> The concept of the Notifer will not work here as a notification message is direct at a single user via the cookie.

Because it uses a `TempData`, but you could create a custom notifier that store the message on application level such as database or server objects .. etc

**Reply by deanmarcussen** [reply]
> > The concept of the Notifer will not work here as a notification message is direct at a single user via the cookie.
> 
> Because it uses a `TempData`, but you could create a custom notifier that store the message on application level such as database or server objects .. etc

Yes Hisham we have this. It's called Layers

**Reply by hishamco** [reply]
Ya, I remember there's an Always layer, which display the layer in all the pages

**Reply by TaylorEdm** [reply]
> The concept of the Notifer will not work here as a notification message is direct at a single user via the cookie.
> 
> I would use a layer widget for this. Found in the Design menu / Widgets

I ended up using a Widget located in a zone that was specified for a specific culture code. This worked very well but I need to figure out dismiss section still. Thank you for your help.
